src/main/scala/com/fulcrumgenomics/vcf/filtration/ReadEndSomaticVariantFilter.scala

```diff
@@ -113,37 +111,110 @@ class EndRepairArtifactLikelihoodFilter(val distance: Int = 2,
     * skewed towards the artifact at very low MAFs, and towards mutations as the MAF increases towards 0.5.
     */
   private[filtration] def priors(pileup: Pileup[PileupEntry], maf: Double) : (Double, Double) = {
-    require(maf >= 0 && maf <= 1, s"Maf must be between 0 and 1. Observed maf ${maf} at ${pileup.refName}:${pileup.pos}.")
+    require(maf >= 0 && maf <= 1, s"Maf must be between 0 and 1. Observed maf $maf at ${pileup.refName}:${pileup.pos}.")
 
     val m = if (maf != 0) maf else {
       logger.warning(s"No alt allele observations found with selected cutoffs at: ${pileup.refName}:${pileup.pos}")
       1.0 / pileup.depth
     }
-
     val priorMutation = min(pow(m * 2, 2), 0.9999)
     val priorArtifact = 1 - priorMutation
     (priorMutation, priorArtifact)
   }
+}
+
+/**
+  * Filter that examines SNVs to see if they are likely end repair fill-in artifacts.
+  *
+  * End repair fill-in artifacts form during the process of filling in single-stranded 3' overhangs to create a blunt end.
+  * Oxidative stress can damage the single-stranded template and induce the formation of DNA lesions, such as
+  * 8-oxoguanine. These modified bases undergo mismatched pairing, which after PCR appear as mutations at the ends of
+  * reads.
+  *
+  * @param distance The distance from the end of the template that defines the region where this
+  *                 artifact may occur
+  * @param pValueThreshold the pvalue at and below which mutations are filtered as being likely
+  *                        caused by the artifact
+  */
+class EndRepairFillInArtifactLikelihoodFilter(override val distance: Int = 15, override val pValueThreshold: Option[Double] = None)
+  extends ReadEndSomaticVariantFilter {
+
+  val Info: VcfInfoHeader = VcfInfoHeader("ERFAP", VcfCount.Fixed(1), VcfFieldType.Float, "P-Value for the End Repair Fill-in Artifact Filter test.")
+  val Filter: VcfFilterHeader = VcfFilterHeader("EndRepairFillInArtifact", "Variant is likely an artifact caused by end repair fill-in.")
+
+  override val VcfInfoLines: Iterable[VcfInfoHeader]     = Seq(Info)
+  override val VcfFilterLines: Iterable[VcfFilterHeader] = Seq(Filter)
+
+  /** Applies only for gt where all alleles are SNVs */
+  def appliesTo(gt: Genotype): Boolean = gt.isHet && gt.calls.forall(_.value.length == 1)
+
+  /** Returns true if the position of the base from the closest read end does not exceed the filter's defined distance. */
+  def isArtifactCongruent(refAllele: Byte, altAllele: Byte, entry : BaseEntry, pileupPosition: Int): Boolean = {
+    val positionInRead       = entry.positionInReadInReadOrder
+    val positionFromOtherEnd = Bams.positionFromOtherEndOfTemplate(entry.rec, pileupPosition)
+
+    // Pick the position that's closest to an end, and test for congruency based on that
+    val pos = {
+      if (positionFromOtherEnd.exists(_ < positionInRead)) positionFromOtherEnd.get
+      else                                                 positionInRead
+    }
```

Comment:
Could also do `(Some(positionInRead) ++ positionFromOtherEnd).min` I think

Reply:
Yes that works! FYI a good amount of this code was copy-pasted from before the PR and I am trying to balance creating too much churn during this refactor for safety sake.

Reply:
Fair enough.